statannotations/statannotations.py

```diff
@@ -448,10 +533,14 @@ def get_box_data(b_plotter, box_name):
         # Increment the counter of annotations in the y_stack array
         y_stack_arr[2, xi1:xi2 + 1] = y_stack_arr[2, xi1:xi2 + 1] + 1
 
-    y_stack_max = max(ymaxs)
+    y_stack_max = max(y_stack_arr[1, :])
+
+    # reset transformation
+    ax_to_data = get_transform_func(ax, 'ax_to_data')
+
     if loc == 'inside':
-        ax.set_ylim((ylim[0], max(1.03 * y_stack_max, ylim[1])))
+        ax.set_ylim(ax_to_data.transform([(0, ylim[0]), (0, max(1.03 * y_stack_max, ylim[1]))])[:, 1])
     elif loc == 'outside':
-        ax.set_ylim((ylim[0], ylim[1]))
+        ax.set_ylim(ax_to_data.transform([(0, ylim[0]), (0, ylim[1])])[:, 1])
```

Comment:
With these longer lines, code would be cleaner if only getting limits was done inside the `if` and `ax.set_ylim` was called below.